metrics2/metrics2/src/main/java/io/helidon/metrics/MetricsSupport.java

```diff
@@ -182,29 +184,40 @@ private void optionsAll(ServerRequest req, ServerResponse res, Registry registry
 
     static String toPrometheusData(Registry... registries) {
         return Arrays.stream(registries)
+                .filter(r -> !r.empty())
                 .map(MetricsSupport::toPrometheusData)
                 .collect(Collectors.joining());
     }
 
     static String toPrometheusData(Registry registry) {
-        return registry.stream()
-                .sorted(Comparator.comparing(Map.Entry::getKey))
-                .collect(StringBuilder::new,
-                        (sb, entry) -> toPrometheusData(sb, entry.getKey(), entry.getValue()),
-                        StringBuilder::append)
-                .toString();
+        StringBuilder builder = new StringBuilder();
+        Set<String> serialized = new HashSet<>();
+        List<Map.Entry<MetricID, HelidonMetric>> metrics = registry.stream()
+                .sorted(Map.Entry.comparingByKey())
+                .collect(Collectors.toList());
+        for (Map.Entry<MetricID, HelidonMetric> entry : metrics) {
```

Comment:
The `List` `metrics` is used only in the immediately next `for` loop. Maybe instead of collecting into the list use `.forEach` after the `.sorted`?